File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        vector<int> answer;
        for(int i=0; i<nums.size(); ++i) {
            if(m.find(nums[i]) != m.end()) {
                answer.push_back(i);
                answer.push_back(m[nums[i]]);
                return answer;
            }
            m[target-nums[i]] = i;
        }
        
        return answer;
    }
};

File: 1-two-sum/1-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let mymap: Map<number, number> = new Map();
    
    
    for(let i: number = 0; i<nums.length; ++i) {
        if(mymap.get(target-nums[i]) != undefined) {
            return [i, mymap.get(target-nums[i])];
        }
        mymap.set(nums[i], i);
    }
};
